packages/flutter/lib/src/widgets/pop_scope.dart

```diff
@@ -49,10 +96,10 @@ import 'routes.dart';
 ///    back gestures in the case of a form with unsaved data.
 ///  * [ModalRoute.registerPopEntry] and [ModalRoute.unregisterPopEntry],
 ///    which this widget uses to integrate with Flutter's navigation system.
-class PopScope extends StatefulWidget {
+class PopScopeWithResult<T> extends StatefulWidget {
```

Comment:
Have you thought about changing these file names to pop_scope_with_result.dart (also the examples)? I guess it's probably overkill, but just wanted to bring it up.